### PR TITLE
Nuxt migration

### PR DESCRIPTION
**Pull request by kennething**
# TL;DR

no wonder everyone hates typescript, nobody knows how to fucking use it
worse implementation of typescript than old regents prep
note: there's 158 ts errors caused by the stores being in options api

# my complaints

| complaint | example of violation | shit-o-meter |
| --- | --- | --- |
| alot of places have //@ts-nocheck or //@ts-ignore (how the code is so shit that @ts-nocheck is needed??) | everywhere | EXTREMELY SHIT |
| explicitly typed refs (explicitly typed the wrong way, too `Ref<string>`) | everywhere | VERY SHIT |
| interfaces arent pascal case (bad) | example of why its bad in calendar.vue line 83 | very shit |
| interfaces dont have jsdoc comments for parameters | utils/types.ts | very shit |
| why are stores written in options api | utils/types.ts | pretty shit |
| props declared wrong | everywhere | pretty shit |
| not typing html typed refs | components/createevent.vue line 120 | pretty shit |
| using constructors instead of types (String() instead of string, Boolean() instead of boolean, etc) | everywhere | pretty shit |
| 1 word variables | everywhere | pretty shit |
| double identical foreach??? | components/checkbox.vue | pretty shit |
| svg comments in the template??? why | components/meetingdetails.vue | kinda shit |
| why is a survey modal in app.vue | app.vue | kinda shit |
| does reset.ts belong in stores? | stores/reset.ts | minor inconvenience |
| inconsistent regular and arrow function declarations | everywhere | minor inconvenience |
| css??? | components/calendar.vue | minor inconvenience |
| unused class | calendar.vue | minor inconvenience |
| page title doesnt change for guidance/statistics | guidance.vue, statsitics.vue | minor inconvenience |
| so many comments and unnecessary comments and discussions in comments like bro use discord | components/calendar.vue line 135 | minor inconvenience |
| explicitly declaring css?? | components/boolean.vue | minor inconvenience |
| not using self closing tags for components | everywhere | meh |
| using Array<T> instead of T[] | everywhere | meh |
| script is above template half the time and below half the time | everywhere | meh |
| using $emit everywhere instead of defining emits | everywhere | meh |
| perfect use case for Record<> | components/calendar.vue line 70 | nitpick |
| using innerhtml instead of using a ref variable or smth 💀 | components/createevent.vue line 151 | nitpick |
| ternary | components/mobile.vue | nitpick |
| guard clause | components/navbar.vue | nitpick |
| deprecated | review.vue line 112 | nitpick |
| no error page (catchall route) | nonexistent | nitpick |
| wtf does string! do | components/createevent.vue line 126 | genuine question |
| why are the svgs in components?? | components/icons | question |
| why are there random pngs in components/icons??? | components/icons | question |
| is a component really necessary if its the only thing on the page | calendar.vue, print.vue  | question |

### CODE REVIEW

**Comment by whydoesthecodenotwork** [comment]
nuh uh
